melange-core/c-decl.dylan

```diff
@@ -1134,7 +1134,7 @@ define method mapped-name
 end method mapped-name;
 
 define method type-name (decl :: <object-declaration>) => (result :: <string>);
-  decl.equated | decl.type.dylan-name;
+  decl.equated | decl.type.type-name;
```

Comment:
Can you tell me about why this change is needed? (And does C-FFI stuff still pass with it?)